### PR TITLE
fix(hub.account): add a banner to invit the customer to update the siret

### PR DESCRIPTION
**Pull request by stif59100**
ref:MANAGER-10205

Signed-off-by: stif59100 <steeve.vanderstocken@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `release/components-w12`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | MANAGER-10205
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

I added a banner to invit the customer to update the siret

### CODE REVIEW

File: packages/manager/apps/hub/src/dashboard/dashboard.controller.js [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+      '#/useraccount/infos?fieldToFocus=siretForm',
+    );
+  }
+
   $onInit() {
+    this.isDisplayableHubBanner.then((data) => {
+      this.availableSiretBanner = data;
+    });
+
+    this.companyNationalIdentificationNumber = this.coreConfig.getUser().companyNationalIdentificationNumber;
```
Comment (ghyenne) [inline packages/manager/apps/hub/src/dashboard/dashboard.controller.js:19]: For the logic you need the same that ACCOUNT_DASHBOARD_URL.

File: packages/manager/apps/hub/src/dashboard/routing.js [hunk 2]
```diff
@@ -51,6 +51,12 @@ export default /* @ngInject */ ($stateProvider, $urlRouterProvider) => {
       hideBreadcrumb: () => true,
       breadcrumb: /* @ngInject */ ($translate) =>
         $translate.instant('manager_hub_dashboard'),
+      isDisplayableHubBanner: /* @ngInject */ (ovhFeatureFlipping) => {
+        const bannerHubSiretId = 'hub:banner-hub-invite-customer-siret';
+        return ovhFeatureFlipping
+          .checkFeatureAvailability(bannerHubSiretId)
+          .then((feature) => feature.isFeatureAvailable(bannerHubSiretId));
+      },
```
Comment (ghyenne) [inline packages/manager/apps/hub/src/dashboard/routing.js:59]:
```suggestion
      isDisplayableHubBanner: /* @ngInject */ (ovhFeatureFlipping) => ovhFeatureFlipping
          .checkFeatureAvailability('hub:banner-hub-invite-customer-siret')
          .then((feature) => feature.isFeatureAvailable(bannerHubSiretId)),
```

**Review by mohammed-zahaf — APPROVED** [review]

**Review by ghyenne — CHANGES_REQUESTED** [review]

**Review by mohammed-zahaf — DISMISSED** [review]

**Review by ghyenne — DISMISSED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

File: packages/manager/apps/hub/src/dashboard/dashboard.controller.js [hunk 3]
```diff
@@ -1,7 +1,44 @@
...
+    });
+
+    this.ACCOUNT_DASHBOARD_URL = this.coreURLBuilder.buildURL(
+      'dedicated',
+      '#/useraccount/infos?fieldToFocus=siretForm',
+    );
+
+    this.companyNationalIdentificationNumber = this.coreConfig.getUser().companyNationalIdentificationNumber;
+    this.legalform = this.coreConfig.getUser().legalform;
+    this.country = this.coreConfig.getUser().country;
```
Comment (ghyenne) [inline packages/manager/apps/hub/src/dashboard/dashboard.controller.js:31]:
```suggestion
    this.availableSiretBanner = false;
    this.isDisplayableHubBanner.then((data) => {
      this.availableSiretBanner = data && !this.user.companyNationalIdentificationNumber && this.user.legalform === 'corporation' && this.user.country === 'FR';
      if (this.availableSiretBanner) {
        this.atInternet.trackPage({
          name: TRACKING_PREFIX,
          type: 'navigation',
        });
      }
    });

    this.ACCOUNT_DASHBOARD_URL = this.coreURLBuilder.buildURL(
      'dedicated',
      '#/useraccount/infos?fieldToFocus=siretForm',
    );
```

File: packages/manager/apps/hub/src/dashboard/dashboard.controller.js [hunk 4]
```diff
@@ -1,7 +1,44 @@
...
+
 export default class DashboardController {
+  /* @ngInject */
+  constructor(coreURLBuilder, coreConfig, atInternet) {
+    this.coreURLBuilder = coreURLBuilder;
+    this.coreConfig = coreConfig;
+    this.atInternet = atInternet;
+    this.TRACKING_PREFIX = TRACKING_PREFIX;
+    this.HIT_PREFIX = HIT_PREFIX;
+  }
```
Comment (ghyenne) [inline packages/manager/apps/hub/src/dashboard/dashboard.controller.js:11]:
```suggestion
  constructor(coreURLBuilder, coreConfig, atInternet) {
    this.coreURLBuilder = coreURLBuilder;
    this.user = this.coreConfig.getUser();
    this.atInternet = atInternet;
    this.TRACKING_PREFIX = TRACKING_PREFIX;
    this.HIT_PREFIX = HIT_PREFIX;
  }
```

File: packages/manager/apps/hub/src/dashboard/dashboard.html [hunk 5]
```diff
@@ -15,6 +15,37 @@
                 tracking-prefix="{{ $ctrl.prefix }}"
             ></ovh-manager-hub-carousel>
         </div>
+        <oui-message
+            type="info"
+            on-dismiss="$ctrl.onDismiss()"
+            data-ng="$ctrl.goToSiret()"
+            dismissable
+            data-ng-if="$ctrl.availableSiretBanner && !$ctrl.companyNationalIdentificationNumber && $ctrl.legalform === 'corporation' && $ctrl.country === 'FR'"
+        >
```
Comment (ghyenne) [inline packages/manager/apps/hub/src/dashboard/dashboard.html:24]:
```suggestion
            data-ng-if="$ctrl.availableSiretBanner"
        >
```

File: packages/manager/apps/hub/src/dashboard/dashboard.html [hunk 6]
```diff
@@ -15,6 +15,37 @@
                 tracking-prefix="{{ $ctrl.prefix }}"
             ></ovh-manager-hub-carousel>
         </div>
+        <oui-message
+            type="info"
+            on-dismiss="$ctrl.onDismiss()"
+            data-ng="$ctrl.goToSiret()"
```
Comment (ghyenne) [inline packages/manager/apps/hub/src/dashboard/dashboard.html:21]: no need.

Comment (ghyenne) [inline packages/manager/apps/hub/src/dashboard/dashboard.html:21]: no need

**Review by ghyenne — APPROVED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

File: packages/manager/apps/hub/src/dashboard/dashboard.component.js [hunk 7]
```diff
@@ -18,6 +18,7 @@ export default {
     services: '<',
     tickets: '<',
     trackingPrefix: '<',
+    isDisplayableHubBanner: '<',
```
Comment (anooparveti) [inline packages/manager/apps/hub/src/dashboard/dashboard.component.js:21]: **nitpick:** `isDisplayableHubBanner` sounds very generic. 
```suggestion
    displaySiretBanner: '<',
```

File: packages/manager/apps/hub/src/dashboard/dashboard.html [hunk 8]
```diff
@@ -15,6 +15,28 @@
                 tracking-prefix="{{ $ctrl.prefix }}"
             ></ovh-manager-hub-carousel>
         </div>
+        <oui-message
+            type="info"
+            on-dismiss="$ctrl.onDismiss()"
+            dismissable
+            data-ng-if="$ctrl.availableSiretBanner"
+        >
+            <span data-translate="manager_hub_dashboard_banner_siret"> </span>
```
Comment (ghyenne) [inline packages/manager/apps/hub/src/dashboard/dashboard.html:24]:
```suggestion
            <span data-translate="manager_hub_dashboard_banner_siret"></span>
```

**Review by ghyenne — DISMISSED** [review]

**Review by anooparveti — CHANGES_REQUESTED** [review]

**Review by ghyenne — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**